js/render/draw_fill.js

```diff
@@ -49,6 +50,39 @@ function draw(painter, source, layer, coords) {
         for (var j = 0; j < coords.length; j++) {
             drawStroke(painter, source, layer, coords[j]);
         }
+    } else if (layer.paint['fill-pattern']) {
```

Comment:
`drawFill` is called twice for each layer each frame: once for the opaque pass and once for the translucent pass. The opaque pass is only used to draw completely opaque geometry. All antialiasing strokes are translucent so they need to be drawn when `painter.isOpaquePass` is false.

No antialiasing should be drawn if `layer.paint['fill-antialias']` is false.

```
if (!painter.isOpaquePass && layer.paint['fill-antialias']) {
    if (strokeColor || !layer.paint['fill-pattern']) {
        // draw outline with either strokeColor or the fill color
    } else {
        // drawn the new patterned outline
    }
}
```